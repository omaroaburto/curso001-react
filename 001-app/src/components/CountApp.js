import React, {useState} from 'react';
import PropTypes from 'prop-types';

const CountApp = ({value}) => {
    //handAdd
    const [count, setCount] = useState(value);
    const handleAdd = () =>{ 
        //forma 1
        setCount(count+1);
        //forma 2
        //setCount((c)=> c+1);
    } 
    const handleSubtract = () =>{ 
        //forma 1
        if(count!==0)
            setCount(count-1); 
    } 
    const handleReset = () =>{ 
        //forma 1
        setCount(value); 
    } 
    return (
        <>
            <h1>CounterApp</h1>
            <span>{count}</span> 
            <br/>
            <button onClick={handleAdd}>+1</button> 
            <button onClick={handleSubtract}>-1</button> 
            <button onClick={handleReset}>Reset</button> 
        </>
    );
}

CountApp.propTypes = {
    value: PropTypes.number
}
CountApp.defaultProps = {
    value: 10
}
export default CountApp
