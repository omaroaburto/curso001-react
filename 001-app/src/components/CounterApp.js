import React from 'react'
import PropTypes from 'prop-types';
//ejemplo de pros
const CounterApp = ({numero}) => {
    return (
        <>
            <h1>Hola mundo</h1>
            <span>{numero}</span>
        </>
    )
}

CounterApp.propTypes={
    numero : PropTypes.number.isRequired
};

export default CounterApp;