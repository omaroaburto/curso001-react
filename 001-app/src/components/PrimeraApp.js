import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({saludo, mensaje}) => {
    return (
        <>
            <h1>{saludo}</h1>
            <span>{mensaje}</span>   
        </>
    )
}

PrimeraApp.propTypes={
    saludo : PropTypes.string.isRequired,
    mensaje: PropTypes.string.isRequired
};

PrimeraApp.defaultProps ={
    saludo: 'Hola gente',
    mensaje: 'Mi nombre es omaro'
}

export default PrimeraApp;