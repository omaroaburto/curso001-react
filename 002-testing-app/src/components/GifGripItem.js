import React from 'react'
import PropTypes from 'prop-types';
export const GifGripItem = ({id, title, url}) => {
    return (
        <div key={id} className="item-caja">
            <img 
                key={id} 
                src={url} 
                alt={title}
            />   
        </div>  
    )
}

GifGripItem.propTypes ={
    title: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired 
}

