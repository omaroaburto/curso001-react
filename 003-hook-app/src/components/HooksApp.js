import React from 'react'

export const HooksApp = () => {
    return (
        <div>
            <h1 className="mt-4">Hola mundo</h1>
        </div>
    )
}
