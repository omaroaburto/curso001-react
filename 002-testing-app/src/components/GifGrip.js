import React from 'react'
import { useFetchGif } from '../hooks/useFetchGif'; 
import { GifGripItem } from './GifGripItem';
import PropTypes from 'prop-types';
export const GifGrip =  ({category}) => { 
    const {data:imagenes, loading} = useFetchGif(category); 

    return (
        <>
            <h3>{category.toUpperCase()}</h3> 
            {loading && <span>Loading...</span>}
            <div className="caja"> 
                {
                    imagenes.map(({url, id, title}) =>{
                            return (
                                <GifGripItem key={Math.random()} url={url} id={id} title={title}/>  
                            );
                    })
                }
            </div> 
        </>
    )
}

GifGrip.propTypes = {
    category: PropTypes.string.isRequired
}
