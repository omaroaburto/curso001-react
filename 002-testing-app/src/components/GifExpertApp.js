import React, {useState} from 'react'
import AddCategory from './AddCategory';
import { GifGrip } from './GifGrip';

export const GifExpertApp = ({elements=['gatos', 'perros']}) => {
    const [categories, setCategories] = useState(elements);

    return (
        <>
          <h1>GifExpertApp</h1>
          <AddCategory setCategories = {setCategories} categories = {categories}/>
          <hr/> 
          <ul>
            {categories.map(category=>{
                return (<GifGrip 
                    category={category}
                    key={category}
                />);
            })}
          </ul>  
        </>
    )
}

 
