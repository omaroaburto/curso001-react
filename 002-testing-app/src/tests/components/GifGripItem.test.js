import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGripItem } from '../../components/GifGripItem'; 

describe('Pruebas del archivo GofGripItem.test.js',()=>{
    
    const id ="S3iIdbIuYNK588i8AT";    
    const title="Gatos GIF by memecandy";
    const url= "https://media2.giphy.com/media/S3iIdbIuYNK588i8AT/giphy.gif?cid=0f518613gtsd2f5q7omh1z014q2wzfl6cj61u2nytsv7bncq&rid=giphy.gif";
    
    const wrapper = shallow(<GifGripItem id={id} title={title} url={url}/>);
    test('Prueba del componente <GifGripItem/>',()=>{ 
        expect(wrapper).toMatchSnapshot();
    });
    test('la imagen necesita un url y un alt',()=>{
        const  img =  wrapper.find('img');
        expect(img.props().src).toBe(url);
        expect(img.props().alt).toBe(title);
    });
    test('Evaluamos si existe la className item-caja',()=>{
        const div =  wrapper.find('div');
        expect(div.props().className).toBe('item-caja');
        
    });
});