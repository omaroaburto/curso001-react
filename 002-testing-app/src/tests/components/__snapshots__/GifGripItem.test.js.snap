// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del archivo GofGripItem.test.js Prueba del componente <GifGripItem/> 1`] = `
<div
  className="item-caja"
  key="S3iIdbIuYNK588i8AT"
>
  <img
    alt="Gatos GIF by memecandy"
    key="S3iIdbIuYNK588i8AT"
    src="https://media2.giphy.com/media/S3iIdbIuYNK588i8AT/giphy.gif?cid=0f518613gtsd2f5q7omh1z014q2wzfl6cj61u2nytsv7bncq&rid=giphy.gif"
  />
</div>
`;
